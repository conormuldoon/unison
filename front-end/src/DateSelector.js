import React from 'react';

import DayPickerInput from 'react-day-picker/DayPickerInput';
import 'react-day-picker/lib/style.css';

import  {
  formatDate,
  parseDate,
} from 'react-day-picker/moment';

import {FORMAT} from './Constant';

/**
 * A component for selecting a date.
 * 
 */
export default function DateSelector(props){


    return (
      <div>
        <p>{props.label}</p>
        <DayPickerInput

          format={FORMAT}
          formatDate={formatDate}
          parseDate={parseDate}
          value={props.dateValue}
          onDayChange={props.handleDayChange}


        />
      </div>
    );

}
