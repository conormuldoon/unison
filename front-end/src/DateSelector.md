DateSelector example:

```js
import {TODAY} from './Util';

<div style={{"width":"80vw"}}>
    <DateSelector label="Select date" dateValue={TODAY} />
</div>
```