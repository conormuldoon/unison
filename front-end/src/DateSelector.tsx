import React from 'react';
import DayPickerInput from 'react-day-picker/DayPickerInput';
import PropTypes from 'prop-types';
import 'react-day-picker/lib/style.css';
import { formatDate, parseDate } from 'react-day-picker/moment';

import { FORMAT } from './Constant';

export interface SelectorProps {

  /**
   * Tha label dispalyed above the date selector.
   */
  label: string;

  /**
   * The initial value of the date selector.
   */
  dateValue: Date;

  /**
   * A callback invoked when the selected date is changed.
   */
  handleDayChange: (day: Date) => void;
}
/**
 * Creates a component for selecting a date.
 * 
 */
function DateSelector({ label, dateValue, handleDayChange }: SelectorProps): JSX.Element {


  return (
    <div>
      <p>{label}</p>
      <DayPickerInput

        format={FORMAT}
        formatDate={formatDate}
        parseDate={parseDate}
        value={dateValue}
        onDayChange={handleDayChange}


      />
    </div>
  );

}

DateSelector.propTypes = {

  /** The label to display over the selector. */
  label: PropTypes.string.isRequired,

  /** The initial date value to display. */
  dateValue: PropTypes.string.isRequired,

  /** A callback function called when the date is changed. */
  handleDayChange: PropTypes.func.isRequired,

}



export default DateSelector;