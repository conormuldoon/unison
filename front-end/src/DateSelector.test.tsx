import { render } from "@testing-library/react";
import DateSelector from './DateSelector';


function handle(d: Date) {

}
const dateSelector = <DateSelector label={'Test Label'} dateValue={'14/9/2021'} handleDayChange={handle} />;

it('renders without crashing', () => {

  render(dateSelector);
});



it('mathes snapshot', () => {


  const { container } = render(dateSelector);

  expect(container).toMatchSnapshot();

});
