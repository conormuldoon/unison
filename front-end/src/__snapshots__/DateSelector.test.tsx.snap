// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <p>
      Test Label
    </p>
    <div
      class="DayPickerInput"
    >
      <input
        placeholder="YYYY-M-D"
        value="14/9/2021"
      />
    </div>
  </div>
</div>
`;
